src/harp-serial/harp/serial/device.py

```diff
@@ -0,0 +1,1347 @@
+            HarpMessage(MessageType.WRITE, PayloadType.U8, address, reg_value)
+        )
+
+        if reply is None:
+            return False
+
+        return reply is not None
+
+    def op_led_en(self, enable: bool) -> bool:
+        """
+        Sets the operation LED of the device.
+
+        Parameters
+        ----------
+        enable : bool
+            If True, enables the operation LED. If False, disables it
+
+        Returns
+        -------
+        bool
+            True if the operation was successful, False otherwise
+        """
+        address = CommonRegisters.OPERATION_CTRL
+
+        # Read register first
+        reply = self.send(HarpMessage(MessageType.READ, PayloadType.U8, address))
+
+        if reply is None:
+            return False
+
+        reg_value = reply.payload
+
+        if enable:
+            reg_value |= OperationCtrl.OPLEDEN
+        else:
+            reg_value &= ~OperationCtrl.OPLEDEN
+
+        reply = self.send(
+            HarpMessage(MessageType.WRITE, PayloadType.U8, address, reg_value)
+        )
+
+        return reply is not None
+
+    def visual_en(self, enable: bool) -> bool:
+        """
+        Sets the status led of the device.
+
+        Parameters
+        ----------
+        enable : bool
+            If True, enables the status led. If False, disables it
+
+        Returns
+        -------
+        bool
+            True if the operation was successful, False otherwise
+        """
+        address = CommonRegisters.OPERATION_CTRL
+
+        # Read register first
+        reply = self.send(HarpMessage(MessageType.READ, PayloadType.U8, address))
+
+        if reply is None:
+            return False
+
+        reg_value = reply.payload
+
+        if enable:
+            reg_value |= OperationCtrl.VISUALEN
+        else:
+            reg_value &= ~OperationCtrl.VISUALEN
+
+        reply = self.send(
+            HarpMessage(MessageType.WRITE, PayloadType.U8, address, reg_value)
+        )
+
+        return reply is not None
+
+    def mute_reply(self, enable: bool) -> bool:
+        """
+        Sets the MUTE_REPLY bit of the device.
+
+        Parameters
+        ----------
+        enable : bool
+            If True, the Replies to all the Commands are muted. If False, un-mutes them
+
+        Returns
+        -------
+        bool
+            True if the operation was successful, False otherwise
+        """
+        address = CommonRegisters.OPERATION_CTRL
+
+        # Read register first
+        reply = self.send(HarpMessage(MessageType.READ, PayloadType.U8, address))
+
+        if reply is None:
+            return False
+
+        reg_value = reply.payload
+
+        if enable:
+            reg_value |= OperationCtrl.MUTE_RPL
+        else:
+            reg_value &= ~OperationCtrl.MUTE_RPL
+
+        reply = self.send(
+            HarpMessage(MessageType.WRITE, PayloadType.U8, address, reg_value)
+        )
+
+        return reply is not None
+
+    def reset_device(
```

Comment:
We should consider removing these syntactic sugar methods and just create message constructors for the core registers instead. Something closer to what we have in the csharp api.

In other words:
```
reset_message = ResetMessage(...)
device.send(reset_message)

```

Seems a bit more scalable in cases where the protocol changes.